server/utils/reviewUtils.ts

```diff
@@ -1,15 +1,16 @@
-import { ApprovedPremisesApplication, ApprovedPremisesAssessment } from '../@types/shared'
+import { ApprovedPremisesApplication as Application, ApprovedPremisesAssessment as Assessment } from '../@types/shared'
 import { SummaryListItem, Task } from '../@types/ui'
 import Apply from '../form-pages/apply'
+import Assess from '../form-pages/assess'
+import isAssessment from './assessments/isAssessment'
 
 const reviewSections = (
-  application: ApprovedPremisesApplication,
-  rowFunction: (
-    task: Task,
-    document: ApprovedPremisesApplication | ApprovedPremisesAssessment,
-  ) => Array<SummaryListItem>,
+  application: Application | Assessment,
+  rowFunction: (task: Task, document: Application | Assessment) => Array<SummaryListItem>,
 ) => {
-  const nonCheckYourAnswersSections = Apply.sections.slice(0, -1)
+  const nonCheckYourAnswersSections = isAssessment(application)
+    ? Assess.sections.slice(1, -1)
```

Comment:
Should we be using [`getSections`](https://github.com/ministryofjustice/hmpps-approved-premises-ui/blob/f303e9bdd0ab6eb2c9e21940b2bfde345902cd56/server/utils/assessmentUtils.ts#L118) here, as if a user proceeds to reject an assessment after not getting a response to an information request (as per https://github.com/ministryofjustice/hmpps-approved-premises-ui/pull/503), then the sections they'll get back will be different.

Reply:
Yeah that makes sense... should I the Matching information to [that filter](https://github.com/ministryofjustice/hmpps-approved-premises-ui/blob/c5430817ddde3c8a96c014980548d35f7af1b4dd/server/utils/assessmentUtils.ts#L122) it doesn't make sense to me to show them Matching Info if the Application has been rejected